base/linalg/lapack.jl

```diff
@@ -31,6 +31,12 @@ macro chkuplo()
       throw(ArgumentError(string("uplo argument must be 'U' (upper) or 'L' (lower), got ", repr(uplo)))))
 end
 
+#Check that {c}transpose is correctly specified
+macro chktrans()
+    :((trans=='N' || trans=='C' || trans=='T') ||
+      throw(ArgumentError(string("trans argument must be 'N' (no transpose), 'T' (transpose), or 'C' (conjugate transpose), got ", repr(trans)))))
+end
```

Comment:
Not sure if function or macro is better but explicit variable name should be perferred unless there's a good reason for not doing that.

There's also a few other macros above which implicitly refering `info`. Might worth changing as well if others do not have trouble with them. (either in this PR or a different one should be fine).